### PR TITLE
avoid importing detect_secrets in setup.py

### LINKED ISSUE #407: detect-secrets 1.0.0 pre-commit hook broken
Hi, and congrats on releasing version 1.0.0 of detect-secrets!  I feel bad filing the first bug report against this new version, but the pre-commit hook / `setup.py` is broken.  I'm using pre-commit version 2.10.1, the latest version.

To reproduce:

1. Have a `.pre-commit-config.yaml` file like this:

```yaml
repos:

  - repo: 'https://github.com/Yelp/detect-secrets'
    rev: 'v1.0.0'
    hooks:
      - id: 'detect-secrets'
```

2. Run `pre-commit run --all-files`.

3. Observe output like this:

```
[INFO] Installing environment for https://github.com/Yelp/detect-secrets.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/Users/manselmi/.cache/pre-commit/repo4vblkajn/py_env-python3/bin/python', '-mpip', 'install', '.')
return code: 1
expected return code: 0
stdout:
    Looking in indexes: http://localhost:3141/root/local/+simple/
    Processing /Users/manselmi/.cache/pre-commit/repo4vblkajn

stderr:
        ERROR: Command errored out with exit status 1:
         command: /Users/manselmi/.cache/pre-commit/repo4vblkajn/py_env-python3/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders
/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-req-build-638l7i5t/setup.py'"'"'; __file__='"'"'/private/var/folders/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-req-build-638l7i5
t/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exe
c'"'"'))' egg_info --egg-base /private/var/folders/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-pip-egg-info-ucyg6p6p
             cwd: /private/var/folders/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-req-build-638l7i5t/
        Complete output (19 lines):
        Traceback (most recent call last):
          File "<string>", line 1, in <module>
          File "/private/var/folders/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-req-build-638l7i5t/setup.py", line 4, in <module>
            from detect_secrets.__version__ import VERSION
          File "/private/var/folders/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-req-build-638l7i5t/detect_secrets/__init__.py", line 1, in <module>
            from .core.secrets_collection import SecretsCollection  # noqa: F401
          File "/private/var/folders/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-req-build-638l7i5t/detect_secrets/core/secrets_collection.py", line 11, in <module>
            from . import scan
          File "/private/var/folders/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-req-build-638l7i5t/detect_secrets/core/scan.py", line 24, in <module>
            from .plugins import Plugin
          File "/private/var/folders/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-req-build-638l7i5t/detect_secrets/core/plugins/__init__.py", line 1, in <module>
            from . import initialize    # noqa: F401
          File "/private/var/folders/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-req-build-638l7i5t/detect_secrets/core/plugins/initialize.py", line 9, in <module>
            from .util import get_mapping_from_secret_type_to_class
          File "/private/var/folders/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-req-build-638l7i5t/detect_secrets/core/plugins/util.py", line 13, in <module>
            from ...plugins.base import BasePlugin
          File "/private/var/folders/tg/06ncct7j4b74t13k0fp8h9sd9w0tnj/T/pip-req-build-638l7i5t/detect_secrets/plugins/base.py", line 19, in <module>
            import requests
        ModuleNotFoundError: No module named 'requests'
        ----------------------------------------
    WARNING: Discarding file:///Users/manselmi/.cache/pre-commit/repo4vblkajn. Command errored out with exit status 1: python setup.py egg_info Check the logs for full
command output.
    ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.

Check the log at /Users/manselmi/.cache/pre-commit/pre-commit.log
```

There is a chicken-and-egg problem in `setup.py`.  `setup.py` imports `detect_secrets` before running the setuptools code that installs runtime dependencies such as requests, and since those dependencies are not yet available, we get a `ModuleNotFoundError`.

### PR DESCRIPTION
**Pull request by cbows**
Fixes #407

### CODE REVIEW

**Review by domanchi — APPROVED** [review]

**Comment by domanchi** [comment]
🤦 Thanks for this, @cbows!

File: setup.py [hunk 1]
```diff
@@ -1,7 +1,15 @@
+from pathlib import Path
 from setuptools import find_packages
 from setuptools import setup
 
-from detect_secrets.__version__ import VERSION
+
+def get_version():
+    """Parse package __version__.py to get version."""
+    versionpy = Path('detect_secrets/__version__.py').read_text()
```
Comment (michael-anselmi) [inline setup.py:8]: This should accommodate platforms that don't use `/` as a path separator:

```suggestion
    versionpy = Path('detect_secrets').joinpath('__version__.py').read_text()
```

Comment (domanchi) [reply setup.py:8]: Fair point. Will ensure this gets added to the patch version.

Comment (cbows) [reply setup.py:8]: not sure if pathlib handles this by default anyway, but a more pythonic way would be to write `Path('detect_secrets') / '__version__.py'`. `/` is actually overloaded for path objects to do the right thing.

**Pull request merged**